### ISSUE TITLE
(1.16) (Linux) the "--prefer-text" parameter does not work

### ISSUE THREAD

**Issue by XayUp** [Genymobile/scrcpy#1920]
To start scrcpy, on linux, I use the following command:
`scrcpy -S -b2M -m800 --prefer-text`
however, "`--prefer-text`" doesn't work, it's like I didn't add it there, so when I want to put an accent, it doesn't work ... for example:
instead of being "ã", he writes "~ a". I don't know if it is a bug or not in this version but this is the first time I use Scrcpy on Linux.

**Comment by rom1v** [comment]
Yes, accented characters are very limited: https://blog.rom1v.com/2018/03/introducing-scrcpy/#handle-accented-characters

It works for `^a` which gives `â` for example.

Ref #1751

**Comment by XayUp** [comment]
(sorry my english) ok ... but is this due to the system? cuz, on windows, the last version I used was 1.14 and the accentuation worked, but ... ok ... this is kind of boring.

**Comment by rom1v** [comment]
Oh if it works on Windows it's probably related to how SDL generates text events.

**Comment by XayUp** [comment]
so, but this in version 1.14 ...

**Comment by XayUp** [comment]
is there no workaround at the moment?

**Comment by rom1v** [comment]
In fact, that's not surprising, on Linux if you press `^a` it generates `â`, but if you press `~a` it generates `~a` (for example, in a terminal).

The opposite would be annoying on Linux (`~` is interpreted as `$HOME` in many shells).

That's probably configurable somewhere.

**Comment by XayUp** [comment]
from what I could see, the "ç" key doesn't work .... I press "ç" but he doesn't write "ç" ... nor does "´" work either ... but I'll look for a solution ... maybe I think
thanks for listening!!

**Comment by rom1v** [comment]
Yes, `ç` is not an ASCII character and cannot be composed as a diacretical character. This is a known limitation of Android text injection (which could be solved by implementing an IME installed as an Android app, but it's not done yet).

Anyway, it is independant of `--prefer-text`, which injects letters as text events instead of key events.

**Comment by XayUp** [comment]
ok .... funny that on windows everything works perfectly ... but that's ok

**Comment by rom1v** [comment]
Oh but now you say it, I think that `ç` works correctly for me on Linux (I'm on the phone so I cannot test).

**Comment by XayUp** [comment]
Oh ;-;

**Comment by rom1v** [comment]
I confirm, `ç` works for me on Linux.

My keyboard layout si `French (alt.)`:

```bash
setxkbmap fr oss
```

Does it work with:

```
adb shell input text ç
```

?

**Comment by XayUp** [comment]
I don't know, friend, but I think I won't depend on it too much, I even use your soft to control a cell phone here that has a broken display

**Comment by vcampitelli** [comment]
I'm using Linux with "English - International with dead keys" keyboard layout and the `--prefer-text` stopped working in version 1.16 for me too. Everything worked correctly when I was using 1.14, but now the behavior is the same with or without the parameter.

Combination | Expected (1.14) | Now (1.16)
------------ | ------------- | -------------
| `'` + `e` | é | e |
| `'` + `c` | ç | c |
| `~` + `a` | ã | ~a |
| `^` + `e` | ê | ^e |


`adb shell input text ç` correctly sends `ç`, but every other letter I wrote on the above table doesn't, unfortunately.

**Comment by rom1v** [comment]
Could you retry v1.14 now and confirms that it still behaves as you expect?

**Comment by vcampitelli** [comment]
@rom1v Yep, I've just reinstalled v1.14 and all combinations work as expected with `--prefer-text`

**Comment by rom1v** [comment]
@vcampitelli Thank you for the details.

I changed the language on my desktop to `English (US, intl., with dead keys)`, and I sometimes get the behavior you describe (but regardless of v1.14 vs v1.16, it seems more related to the fact that the virtual keyboard is visible when I type or not yet).

If you can reproduce the problem 100% on v1.16 and 0% on v1.14, could you `git bisect` to find the commit where the problem appears, please?

**Comment by rom1v** [comment]
Also, what's your Android keyboard?

**Comment by vcampitelli** [comment]
@rom1v sorry for the late reply. I'm using the default Gboard app.

So, the strangest thing happened. I [manually build scrcpy](https://github.com/Genymobile/scrcpy/blob/master/BUILD.md) using several commits and everything works! But I don't know why, actually.

I used the same dependencies versions that comes with the snap package _(SDL 2.0.8, libavcodec 57.107.100, libavformat 57.83.100 and libavutil 55.78.100)_ and used docker _(my own image instead of the one provided by [pierlon/scrcpy-docker](https://github.com/pierlon/scrcpy-docker))_. Maybe another dependency or some kind of environment configuration while building the package?

Until I figure it out, I'm going to use the one I build instead of the package. Thanks!

@XayUp maybe you should do it too, it's very simple!

**Comment by 1049731887** [comment]
And cannot input any Chinese words, but it can be input in Vysor if you turn on the button of "International Keyboard"
